drv/sprot-api/README.md

```diff
@@ -0,0 +1,510 @@
+subargs=SpRotArgs { send: None, pulse: None, sink: None, status: true, msgtype: "echoreq", timeout: 5000 }
+ops=[Call(TargetFunction(34)), Done]
+empty data: None
+status=Status {
+    supported: 0x00000002,
+    bootrom_crc32: 0x47ae8b8d,
+    rx_invalid: 0x00000000,
+    rx_nop: 0x00000001,
+    rx_overrun: 0x00000012,
+    rx_received: 0x000001af,
+    tx_incomplete: 0x00000000,
+    tx_underrun: 0x00000013,
+}
+```
+
+The ErrorReq/ErrorRsp and SinkReq/SinkRsp messages are not expected to be useful in production and should be conditionally compiled or removed.
+
+The returned SinkRsp message contains a SpRotSinkStatus struct:
+
+```rust
+#[repr(C, packed)]
+pub struct SpRotSinkStatus {
+    pub sent: u16,
+    pub req_crc_err: u16,
+    pub rsp_crc_err: u16,
+    pub flow_err: u16,
+}
+```
+
+So in the above test, 200 messages were sent by the RoT (`SinkRsp`) and there
+were 15 flow errors. There were no CRC errors.
+
+The test code in the SP successfully retried those 15 messages and returned `Ok`
+for the test.
+
+#### Test StatusReq and StatusRsp:
+
+The exact values in the status structure are expected to change and stabilize
+before first customer ship.
+
+/Discussion on contents are appreciated/.
+
+  $ cargo xtask humility $APP -- sprot --status
+...
+status=Status {
+    supported: 0x00000002,
+    bootrom_crc32: 0x47ae8b8d,
+    rx_invalid: 0x00000000,
+    rx_nop: 0x00000001,
+    rx_overrun: 0x00000001,
+    rx_received: 0x000000cd,
+    tx_incomplete: 0x00000000,
+    tx_underrun: 0x00000001,
+}
+
+```
+
+#### Test sprockets messages from laptop RS232 to SP to RoT and back:
+
+Connect a USB to RS232 adaptor, such as the "[OIKWAN USB Serial Adapter with FTDI Chipset](https://www.amazon.com/gp/product/B0759HSLP1)"
+
+A "[9 Pin Serial Male to 10 Pin Motherboard Header Panel Mount Cable](https://www.amazon.com/StarTech-com-Serial-Motherboard-Header-Panel/dp/B0067DB6RU)" makes simple work of the connection. With these particular cables, a female/female DB9 adapter or Dupont wires between pins 2, 3, and 5 are needed to finish the connections.
+
+Alternatively, use dupont wires directly to the Gemini J302 connector:
+
+Use dupont female to female wires to connect pins:
+  - `DB9:2(RXD)` to `J302:3(RS232_TX)`
+  - `DB9:3(TXD)` to `J302:5(RS232_RX)`
+  - `DB9:5(GND)` to `J302:9(GND)`
+
+DB9, looking at the pins:
+```
+ ___________
+( 1 2 3 4 5 )
+ \ 6 7 8 9 /
+  \_______/
+```
+
+Gemini connector J302 labeled:
+
+```
+        "SERIAL TO HOST"
+    (TODO NOTE SIGNAL LEVELS)
+       +-----------------+
+J302   |  2  4  6  8 10  |
+       |  1  3  5  7  9  |
+       +-------   -------+
+```
+
+Identify the device path for your USB to RS232 adaptor. On the author's Linux laptop, it shows up as `/dev/ttyUSB0`.
+
+```sh
+SERIAL=/dev/ttyUSB0
+```
+
+In a [sprockets workspace](https://github.com/oxidecomputer/sprockets):
+
+```sh
+  cargo run -- -b 115200 -p $SERIAL get-certificates
+  cargo run -- -b 115200 -p $SERIAL get-measurements
```

Comment:
I don't see any uart config / setup in this PR - how is this making its way to sprot?